### PR TITLE
LPD-36188 Root PR

### PR DESCRIPTION
**Pull request by izaera**
This draft PR contains all commits for [LPD-36188](https://liferay.atlassian.net/browse/LPD-36188).

We are tracking the progress of teams' sub-PRs at [LPD-36186](https://liferay.atlassian.net/browse/LPD-36186).

The "Welcome Pack" (in Technical Draft format) for teams' sub-PRs is at this [confluence doc](https://liferay.atlassian.net/wiki/spaces/ENGFRONTENDINFRA/pages/3245277533/LPD-36188+-+Allow+using+directive+script-src-attr+none+by+removing+inline+handlers).

I will split it in parts based on ownership and will send sub-PRs accordingly for CI testing.

This PR reflects the real time status of LPD-36188 using emojis, like this:

- :see_no_evil: Infrastructure commits that cannot really be tested directly (because they affect the majority of the portal)
- :ok_hand: Commits for which we know what test suite to run (based on heuristics, see below)
- :link: Commits for which we know which team we should send the PR to (based on heuristics, see below)
- :question: Commits for which we don't yet know the team or the test suite
- :checkered_flag:  Commit that has already been submitted in a sub-PR

**How to determine test suite and team for a given commit:**
1. Determine the "app" to which the commit belongs (eg: `modules/apps/wiki`).
2. Perform a `git log` on that app directory.
3. Find a meaningful LPD and based on the user and the PR, determine:
    1. The `liferay-portal` fork for the app (eg: `https://github.com/liferay-content-management`)
    2. The typical test suite used by that team (eg: `ci:test:content-management`)

**List of teams affected by this PR and their ongoing sub-PRs:**
1. ~https://github.com/liferay-appsec/liferay-portal :point_right: https://github.com/liferay-frontend/liferay-portal/pull/4468~ (SENT :white_check_mark:) 
2. ~https://github.com/liferay-commerce/liferay-portal :point_right: https://github.com/liferay-frontend/liferay-portal/pull/4471~ (SENT :white_check_mark:) 
3. ~https://github.com/liferay-content-management/liferay-portal :point_right: https://github.com/liferay-frontend/liferay-portal/pull/4473~ (SENT :white_check_mark:)
4. ~https://github.com/liferay-forms/liferay-portal :point_right: https://github.com/liferay-frontend/liferay-portal/pull/4474~ (SENT :white_check_mark:)
5. ~https://github.com/liferay-frontend/liferay-portal :point_right: https://github.com/liferay-frontend/liferay-portal/pull/4475~ (MERGED :white_check_mark:)
6. ~https://github.com/liferay-headless/liferay-portal :point_right: https://github.com/liferay-frontend/liferay-portal/pull/4476~ (SENT :white_check_mark:)
7. ~https://github.com/liferay-search/liferay-portal :point_right: https://github.com/liferay-frontend/liferay-portal/pull/4477~ (SENT :white_check_mark:)
8. ~https://github.com/liferay-content-management/liferay-portal :point_right: https://github.com/liferay-frontend/liferay-portal/pull/4479~ (SENT :white_check_mark:)

**Confluence doc to attach to each team PR explaining what's going on:** 
https://liferay.atlassian.net/wiki/spaces/ENGFRONTENDINFRA/pages/3245277533/LPD-36188+-+Allow+using+directive+script-src-attr+none+by+removing+inline+handlers

**Rest of :see_no_evil: commits:**

I've analyzed where each commit subject is used and have come to these conclusions:

```
=== Used pervasively:
- RowChecker
- <aui:a>
- <aui:button>
- <aui:input>
	- <liferay-ui:input-resource> (bc it is only used in <aui:input>)
	- <liferay-ui:input-checkbox> (bc it is mainly used in <aui:input>)
- <aui:select>
- <clay:alert>
- <liferay-ui:input-permissions>
- <liferay-ui:search-container-row>
	- <liferay-ui:page-iterator> (bc it is mainly used in <liferay-ui:search-paginator>)
- <liferay-ui:tabs>

=== Not used at all:
- <liferay-ui:quick-access>
```

So I will create 9 :see_no_evil: (high risk) PRs and one more with `<liferay-ui:quick-access>` that I will send ASAP since it is not expected to break anything.

**Back to square one (Oct 10th):**

Since [duplicated code has been found in revisions](https://github.com/brianchandotcom/liferay-portal/pull/155357#issuecomment-2423742140) we are going to change the approach of this PR and try to use `<aui:..>` tags instead of raw HTML to hide the id computation and attachment inside their implementation.

We will find a better way to do all this under [LPD-39791](https://liferay.atlassian.net/browse/LPD-39791), then undo everything and redo it with the new approach.

### CODE REVIEW

**Review by bryceosterhaus — COMMENTED** [review]
bless your heart, this looks like terribly tedious work...

My only concern is the random id parts, I'm not confident that those will always be unique and it could possibly render the same id twice on the page.

File: modules/apps/comment/comment-taglib/src/main/resources/META-INF/resources/discussion/view_message_thread.jsp [hunk 1]
```diff
@@ -210,9 +210,20 @@ Format dateTimeFormat = FastDateFormatFactoryUtil.getDateTime(locale, timeZone);
 							<c:if test="<%= !discussion.isMaxCommentsLimitExceeded() %>">
 								<c:choose>
 									<c:when test="<%= commentTreeDisplayContext.isReplyButtonVisible() %>">
-										<button class="btn btn-outline-borderless btn-outline-secondary btn-sm" onclick="<%= randomNamespace %>showPostReplyEditor(<%= index %>);" type="button">
+
+										<%
+										String id = StringUtil.randomId();
```
[the remark this answers is not in the document]
Comment (izaera) [reply modules/apps/comment/comment-taglib/src/main/resources/META-INF/resources/discussion/view_message_thread.jsp:215]: It's true they could collide but we use `randomId` all over DXP so it's not a problem just of this PR.

In any case `randomId` uses 4 random letters which means 27^4 possibilities. And ChatGPT :sweat_smile:  says that is 531441 so small enough I guess.

We can use longer ids if we think it makes sense... :thinking:

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/render/TextHTMLDDMFormFieldValueRenderer.java [hunk 2]
```diff
@@ -34,21 +36,31 @@ protected ValueAccessor getValueAccessor(Locale locale) {
...
+				sb.append(")</a><script");
+				sb.append(
+					ContentSecurityPolicyNonceProviderUtil.getNonceAttribute(
+						null));
+				sb.append(">document.getElementById('");
+				sb.append(id);
+				sb.append("').onclick=function() {Liferay.DDLUtil.");
+				sb.append("openPreviewDialog('");
+				sb.append(HtmlUtil.escapeJS(value.getString(locale)));
+				sb.append("');}</script>");
```
[the remark this answers is not in the document]
Comment (izaera) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/render/TextHTMLDDMFormFieldValueRenderer.java:58]: SF

File: modules/apps/frontend-js/frontend-js-bundle-config-extender/src/main/java/com/liferay/frontend/js/bundle/config/extender/internal/JSBundleConfigServlet.java [hunk 3]
```diff
@@ -82,7 +82,7 @@ protected void service(
 			JSBundleConfigRegistryUtil.getJSConfigs();
 
 		if (!jsConfigs.isEmpty()) {
-			printWriter.print("(function(){");
+			printWriter.print("(function() {");
```
[the remark this answers is not in the document]
Comment (izaera) [reply modules/apps/frontend-js/frontend-js-bundle-config-extender/src/main/java/com/liferay/frontend/js/bundle/config/extender/internal/JSBundleConfigServlet.java:85]: SF too :shrug:

**Pull request closed without merging**